### ISSUE TITLE
Using `class_name` before `extends` causes the base class to be wrongly registered

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#24364]
Godot 3.1 master ca28c455bfdc8408485c217c17f07011c0b43f64

While implementing #19501, I found that if you do this:

```gdscript
tool
class_name AlpacaSaver
extends ResourceFormatSaver
```

The class gets registered with a base class `Reference` instead of `ResourceFormatSaver`.
But, if I reorder lines this way, it works:

```gdscript
tool
extends ResourceFormatSaver
class_name AlpacaSaver
```

**Comment by akien-mga** [comment]
It should be made to either work regardless of the order, or, if not possible, a syntax error to have `class_name` before `extends`.

**Comment by akien-mga** [comment]
Fixed by #27987.
